tests/neuro_san/client/test_hocons.py

```diff
@@ -0,0 +1,49 @@
+# Copyright (C) 2023-2025 Cognizant Digital Business, Evolutionary AI.
+# All Rights Reserved.
+# Issued under the Academic Public License.
+#
+# You can be released from the terms, and requirements of the Academic Public
+# License by purchasing a commercial license.
+# Purchase of a commercial license is mandatory for any use of the
+# neuro-san SDK Software in commercial settings.
+#
+# END COPYRIGHT
+
+from unittest import TestCase
+
+from parameterized import parameterized
+
+from tests.framework.unittest.dynamic_hocon_unit_tests import DynamicHoconUnitTests
+
+
+class TestHocons(TestCase):
+    """
+    Data-driven dynamic test cases where each test case is specified by a single hocon file.
+    """
+
+    # A single instance of the DynamicHoconUnitTests helper class.
+    # We pass it our source file location and a relative path to the common
+    # root of the test hocon files listed in the @parameterized.expand()
+    # annotation below so the instance can find the hocon test cases listed.
+    DYNAMIC = DynamicHoconUnitTests(__file__, path_to_basis="../../fixtures")
+
+    @parameterized.expand(DynamicHoconUnitTests.from_hocon_list([
+        "math_guy/basic_sly_data.hocon",
+        "math_guy/forwarded_sly_data.hocon",
+        "music_nerd/beatles_with_history.hocon",
```

Comment:
The list of hocon test case data files that supersedes what was in test_math_guy and test_music_nerd.

These get translated in to test case names as well. Pytest output looks like this:

```
tests/neuro_san/client/test_hocons.py::TestHocons::test_hocon_0_math_guy_basic_sly_data PASSED                   [ 33%]
tests/neuro_san/client/test_hocons.py::TestHocons::test_hocon_1_math_guy_forwarded_sly_data PASSED               [ 66%]
tests/neuro_san/client/test_hocons.py::TestHocons::test_hocon_2_music_nerd_beatles_with_history PASSED           [100%]
```